From e33bee7ceab5e5f8bfbd01c3d26c60f463cc0088 Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Thu, 19 Sep 2019 23:40:49 +0200
Subject: [PATCH 1/4] Update desktop file

 - Update Version entry (Keyword key was introduced in 1.1)
 - Fix GenericName
 - Add TryExec entry
 - Update Keywords
---
 rtdata/rawtherapee.desktop.in | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/rtdata/rawtherapee.desktop.in b/rtdata/rawtherapee.desktop.in
index 233a737b6..9356c5a2d 100644
--- a/rtdata/rawtherapee.desktop.in
+++ b/rtdata/rawtherapee.desktop.in
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Type=Application
-Version=1.0
+Version=1.1
 Name=RawTherapee
-GenericName=Raw photo editor
+GenericName=Raw Photo Editor
 GenericName[cs]=Editor raw obrázků
 GenericName[fr]=Éditeur d'images raw
 GenericName[pl]=Edytor zdjęć raw
@@ -11,9 +11,10 @@ Comment[cs]=Program pro konverzi a zpracování digitálních raw fotografií
 Comment[fr]=Logiciel de conversion et de traitement de photos numériques de format raw (but de capteur)
 Comment[pl]=Zaawansowany program do wywoływania zdjęć typu raw
 Icon=rawtherapee
+TryExec=rawtherapee
 Exec=rawtherapee %f
 Terminal=false
 MimeType=image/jpeg;image/png;image/tiff;image/x-adobe-dng;image/x-canon-cr2;image/x-canon-crf;image/x-canon-crw;image/x-fuji-raf;image/x-hasselblad-3fr;image/x-hasselblad-fff;image/x-jpg;image/x-kodak-dcr;image/x-kodak-k25;image/x-kodak-kdc;image/x-leaf-mos;image/x-leica-rwl;image/x-mamiya-mef;image/x-minolta-mrw;image/x-nikon-nef;image/x-nikon-nrw;image/x-olympus-orf;image/x-panasonic-raw;image/x-panasonic-rw2;image/x-pentax-pef;image/x-pentax-raw;image/x-phaseone-iiq;image/x-raw;image/x-rwz;image/x-samsung-srw;image/x-sigma-x3f;image/x-sony-arq;image/x-sony-arw;image/x-sony-sr2;image/x-sony-srf;image/x-tif;
 Categories=Photography;Graphics;2DGraphics;RasterGraphics;GTK;
-Keywords=raw;photography;develop;pp3;graphics;
+Keywords=raw;photo;photography;develop;pp3;graphics;
 StartupWMClass=rawtherapee

From 633eab24c288b2a8933af23b88d057f7df2ed0f3 Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Thu, 19 Sep 2019 23:44:17 +0200
Subject: [PATCH 2/4] Update AppData file

 - Fix description (<p> tag)
 - Remove keywords (should be obtained from desktop file)
 - Fix URLs
 - Fix screenshot caption (too long)
---
 com.rawtherapee.RawTherapee.appdata.xml | 28 +++++++------------------
 1 file changed, 7 insertions(+), 21 deletions(-)

diff --git a/com.rawtherapee.RawTherapee.appdata.xml b/com.rawtherapee.RawTherapee.appdata.xml
index 0e5dfef50..564821cb4 100644
--- a/com.rawtherapee.RawTherapee.appdata.xml
+++ b/com.rawtherapee.RawTherapee.appdata.xml
@@ -8,31 +8,17 @@
     <summary xml:lang="pl">Zaawansowany program do wywoływania zdjęć typu raw</summary>
     <translation type="gettext">rawtherapee</translation>
     <description>
-        <p>
-            RawTherapee is a powerful, cross-platform raw photo processing program. It is written mostly in C++ using a GTK+ front-end. It uses a patched version of dcraw for reading raw files, with an in-house solution which adds the highest quality support for certain camera models unsupported by dcraw and enhances the accuracy of certain raw files already supported by dcraw. It is notable for the advanced control it gives the user over the demosaicing and development process.
-        </p>
-        <p>
-            RawTherapee is designed for developing raw files from a broad range of digital cameras, as well as HDR DNG files and non-raw image formats (JPEG, TIFF and PNG). The target audience ranges from enthusiast newcomers who wish to broaden their understanding of how digital imaging works to semi-professional photographers. Knowledge in color science is not compulsory, but it is recommended that you are eager to learn and ready to read our documentation (RawPedia) as well as look up basic concepts which lie outside the scope of RawPedia, such as color balance, elsewhere.
-        </p>
-        <p>
-            Of course, professionals may use RawTherapee too while enjoying complete freedom, but will probably lack some peripheral features such as Digital Asset Management, printing, uploading, etc. RawTherapee is not aimed at being an inclusive all-in-one program, and the open-source community is sufficiently developed by now to offer all those peripheral features in other specialized software.
-        </p>
+        <p>RawTherapee is a powerful, cross-platform raw photo processing program. It is written mostly in C++ using a GTK+ front-end. It uses a patched version of dcraw for reading raw files, with an in-house solution which adds the highest quality support for certain camera models unsupported by dcraw and enhances the accuracy of certain raw files already supported by dcraw. It is notable for the advanced control it gives the user over the demosaicing and development process.</p>
+        <p>RawTherapee is designed for developing raw files from a broad range of digital cameras, as well as HDR DNG files and non-raw image formats (JPEG, TIFF and PNG). The target audience ranges from enthusiast newcomers who wish to broaden their understanding of how digital imaging works to semi-professional photographers. Knowledge in color science is not compulsory, but it is recommended that you are eager to learn and ready to read our documentation (RawPedia) as well as look up basic concepts which lie outside the scope of RawPedia, such as color balance, elsewhere.</p>
+        <p>Of course, professionals may use RawTherapee too while enjoying complete freedom, but will probably lack some peripheral features such as Digital Asset Management, printing, uploading, etc. RawTherapee is not aimed at being an inclusive all-in-one program, and the open-source community is sufficiently developed by now to offer all those peripheral features in other specialized software.</p>
     </description>
-    <keywords>
-        <keyword>raw</keyword>
-        <keyword>photo</keyword>
-        <keyword>photography</keyword>
-        <keyword>develop</keyword>
-        <keyword>pp3</keyword>
-        <keyword>graphics</keyword>
-    </keywords>
     <content_rating type="oars-1.0" />
     <metadata_license>CC-BY-SA-4.0</metadata_license>
     <project_license>GPL-3.0+</project_license>
-    <url type="bugtracker">https://github.com/Beep6581/RawTherapee/issues/new</url>
+    <url type="bugtracker">https://github.com/Beep6581/RawTherapee/issues</url>
     <url type="donation">https://www.paypal.me/rawtherapee</url>
-    <url type="help">https://rawpedia.rawtherapee.com/</url>
-    <url type="homepage">https://www.rawtherapee.com/</url>
+    <url type="help">https://rawpedia.rawtherapee.com</url>
+    <url type="homepage">https://www.rawtherapee.com</url>
     <url type="translate">https://discuss.pixls.us/t/localization-how-to-translate-rawtherapee-and-rawpedia/2594</url>
     <launchable type="desktop-id">rawtherapee.desktop</launchable>
     <releases>
@@ -48,7 +34,7 @@
     </provides>
     <screenshots>
         <screenshot type="default">
-            <caption>Color-correcting a drosera rotundifolia in RawTherapee 5.7.</caption>
+            <caption>Color-correcting a drosera rotundifolia</caption>
             <image width="1920" height="1080">https://rawtherapee.com/images/screenshots/rt570_1.jpg</image>
         </screenshot>
         <screenshot>

From 2415511b7011434b6f088fb46df8f4737b9e717b Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Thu, 19 Sep 2019 23:55:31 +0200
Subject: [PATCH 3/4] Update desktop file

 - Change order of categories (Main Category before Additional Categories)
---
 rtdata/rawtherapee.desktop.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtdata/rawtherapee.desktop.in b/rtdata/rawtherapee.desktop.in
index 9356c5a2d..c6c675c4d 100644
--- a/rtdata/rawtherapee.desktop.in
+++ b/rtdata/rawtherapee.desktop.in
@@ -15,6 +15,6 @@ TryExec=rawtherapee
 Exec=rawtherapee %f
 Terminal=false
 MimeType=image/jpeg;image/png;image/tiff;image/x-adobe-dng;image/x-canon-cr2;image/x-canon-crf;image/x-canon-crw;image/x-fuji-raf;image/x-hasselblad-3fr;image/x-hasselblad-fff;image/x-jpg;image/x-kodak-dcr;image/x-kodak-k25;image/x-kodak-kdc;image/x-leaf-mos;image/x-leica-rwl;image/x-mamiya-mef;image/x-minolta-mrw;image/x-nikon-nef;image/x-nikon-nrw;image/x-olympus-orf;image/x-panasonic-raw;image/x-panasonic-rw2;image/x-pentax-pef;image/x-pentax-raw;image/x-phaseone-iiq;image/x-raw;image/x-rwz;image/x-samsung-srw;image/x-sigma-x3f;image/x-sony-arq;image/x-sony-arw;image/x-sony-sr2;image/x-sony-srf;image/x-tif;
-Categories=Photography;Graphics;2DGraphics;RasterGraphics;GTK;
+Categories=Graphics;Photography;2DGraphics;RasterGraphics;GTK;
 Keywords=raw;photo;photography;develop;pp3;graphics;
 StartupWMClass=rawtherapee

From 581e46c7b053b69ebc488011f43e7290870fa287 Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Thu, 19 Sep 2019 23:59:00 +0200
Subject: [PATCH 4/4] Fix AppData file

 - Remove unreachable screenshot URL
---
 com.rawtherapee.RawTherapee.appdata.xml | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/com.rawtherapee.RawTherapee.appdata.xml b/com.rawtherapee.RawTherapee.appdata.xml
index 564821cb4..41a669532 100644
--- a/com.rawtherapee.RawTherapee.appdata.xml
+++ b/com.rawtherapee.RawTherapee.appdata.xml
@@ -33,10 +33,6 @@
         <binary>rawtherapee-cli</binary>
     </provides>
     <screenshots>
-        <screenshot type="default">
-            <caption>Color-correcting a drosera rotundifolia</caption>
-            <image width="1920" height="1080">https://rawtherapee.com/images/screenshots/rt570_1.jpg</image>
-        </screenshot>
         <screenshot>
             <caption>HDR DNG of a misty morning in the countryside</caption>
             <image width="1920" height="1080">https://rawtherapee.com/images/screenshots/rt540_1.jpg</image>
